extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisOutcomePanels.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { VSCodeBadge, VSCodePanels, VSCodePanelTab, VSCodePanelView } from '@vscode/webview-ui-toolkit/react';
+import { formatDecimal } from '../../pure/number';
+import { VariantAnalysis } from '../../remote-queries/shared/variant-analysis';
+import { VariantAnalysisAnalyzedRepos } from './VariantAnalysisAnalyzedRepos';
+import { VariantAnalysisNotFoundRepos } from './VariantAnalysisNotFoundRepos';
+import { VariantAnalysisNoCodeqlDbRepos } from './VariantAnalysisNoCodeqlDbRepos';
+import { Alert } from '../common';
+
+export type VariantAnalysisOutcomePanelProps = {
+  variantAnalysis: VariantAnalysis;
+};
+
+const Tab = styled(VSCodePanelTab)`
+  text-transform: uppercase;
+`;
+
+const WarningsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 1em;
+
+  margin-top: 1em;
+
+  > * {
+    // Add a margin to the last alert, independent of the number of alerts. This will not add a margin when
+    // there is no warning to ensure we do not have a margin-top AND a margin-bottom.
+    &:last-child {
+      margin-bottom: 1em;
+    }
+  }
+`;
+
+export const VariantAnalysisOutcomePanels = ({
+  variantAnalysis
+}: VariantAnalysisOutcomePanelProps) => {
+  const noCodeqlDbRepositoryCount = variantAnalysis.skippedRepos?.noCodeqlDbRepos?.repositoryCount ?? 0;
+  const notFoundRepositoryCount = variantAnalysis.skippedRepos?.notFoundRepos?.repositoryCount ?? 0;
+  const overLimitRepositoryCount = variantAnalysis.skippedRepos?.overLimitRepos?.repositoryCount ?? 0;
+  const accessMismatchRepositoryCount = variantAnalysis.skippedRepos?.accessMismatchRepos?.repositoryCount ?? 0;
+
+  const warnings = (
+    <WarningsContainer>
+      {overLimitRepositoryCount > 0 && (
+        <Alert
+          type="warning"
+          title="Repository limit exceeded"
+          message={`The number of requested repositories exceeds the maximum number of repositories supported by multi-repository variant analysis. ${overLimitRepositoryCount} ${overLimitRepositoryCount === 1 ? 'repository was' : 'repositories were'} skipped.`}
```

Comment:
Maybe I'm just missing them but I can't see this warning or the "access mismatch" warning as part of the designs. Do we need general warning above the tabs, or is it enough just to have the tabs and then there are warnings inside that explain why repos were skipped?

We can leave the implementation of the warnings inside the tabs for when implementing those tabs.

Reply:
We talked about this and I was getting confused between the different error states. These general warnings are for the "too many repos" and "public controller repo" cases, which are different from the "no access" and "no db" cases which get their own tabs. So these warnings are all good.

Regarding the specific text for them, let's go with this for now as just a placeholder while everything is private. We'll have a discussion offline about what text to use here and in general throughout the extension.